components/model_analyzer/TorchBenchAnalyzer.py

```diff
@@ -0,0 +1,120 @@
+        self.gpus = self.gpu_factory.verify_requested_gpus(['all', ])
+        # the metrics to be collected
+        # self.gpu_metrics = [GPUUtilization, GPUPowerUsage,
+        #                     GPUFreeMemory, GPUUsedMemory, GPUFP32Active, GPUTensorActive]
+        self.gpu_metrics = [GPUFP32Active]
+        # the final metric results. Its format is {GPU_UUID: {GPUUtilization: }}
+        self.gpu_metric_value = {}
+        self.gpu_monitor = None
+        self.gpu_records = None
+        self.config = AnalayzerConfig()
+        set_logger()
+
+    def start_monitor(self):
+        try:
+            self.gpu_monitor = DCGMMonitor(
+                self.gpus, self.config.monitoring_interval, self.gpu_metrics)
+            self.gpu_monitor.start_recording_metrics()
+        except TorchBenchAnalyzerException:
+            self._destory_monitor()
+            raise
+
+    def _destory_monitor(self):
+        self.gpu_monitor.destroy()
+        self.gpu_monitor = None
+    
+    def stop_monitor(self):
+        """
+        @return: collected gpu records @todo: add return type 
+        """
+        self.gpu_records = self.gpu_monitor.stop_recording_metrics()
+        self._destory_monitor()
+        return self.gpu_records
+    
+    def aggregate(self):
+        gpu_record_aggregator = RecordAggregator()
+        gpu_record_aggregator.insert_all(self.gpu_records)
+        # @Yueming Hao todo: groupby GPU, then metric. Then calculate the flops
+        records_groupby_gpu = {}
+        records_groupby_gpu = gpu_record_aggregator.groupby(
+            self.gpu_metrics, lambda record: record.device_uuid())
+        
+        for gpu in self.gpus:
+            self.gpu_metric_value[gpu.device_uuid()] = {}
+        for metric_type, metric in records_groupby_gpu.items():
+            for gpu_uuid, metric_value in metric.items():
+                self.gpu_metric_value[gpu_uuid][metric_type] = metric_value
+        # return self.gpu_metric_value
+    
+    def set_monitoring_interval(self, attempted_interval):
+        """
+        The default monitoring internval is 10ms.
+        """
+        if attempted_interval < 0.001:
+            logger.warning("The attempted interval is too short, would cause untruested profiling results.")
+        self.config.monitoring_interval = attempted_interval
+
+    def print_flops(self):
+        print("==========Summary==========")
+        for gpu_uuid in self.gpu_metric_value:
+            gpu = self.gpu_factory.get_device_by_uuid(gpu_uuid)
+            print(self.gpu_metric_value[gpu_uuid][GPUFP32Active].value())
+            # TFLOPs/second = Device_SM_Count x Device_FMAs_Per_Cycle_Per_SM x 2 x Running_Frequency_KHz x DCGM_Activity / 1e+9
+            print("GPU : TFLOPs/Second %.4f" % (gpu._sm_count * gpu._fma_count * 2 *
+                gpu._frequency * self.gpu_metric_value[gpu_uuid][GPUFP32Active].value() / 1e+9))
+        # @Yueming Hao: print all collected gpu records, for debug only
```

Comment:
It will be better to print full values when user set this to debug mode, say, `print_flops(self, debug=False)` and print full information when `debug=True`.